### ISSUE TITLE
SemanticsNode id generator should be reset between tests

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#11791]
If you have two tests in the same file, the first SemanticsNode generated in the second test will have as an id the id of the last node in the previous test plus 1. That's problematic as quite a few semantics tests currently assert on those ids. However, if tests are started to be executed in different order, those tests would fail.
